### DISCUSSION TITLE
Slow Library Scanning

### DISCUSSION THREAD

**Discussion by aransou** [jellyfin/jellyfin#7249] [Q&A]
Hi*,

I'm new user of this software. My setup is a Raspberry 4/4GB Ram and the jellyfin server (10.8.0 Alpha 5) in a fresh docker enviroment. The media files are stored on the host on a hdd. 

I tried a test scenario with round about 10.000 .mp3 files and it tooks multiple hours to initial scan this folder. How can I check i there is a problem with the file IO? (I tried also the 10.7 with seam effect)

CPU load 5-10%, no other IO topic ongoing.

Is there a debug output that I can switch on to see what needs to much time?

**Comment by s-h-a-d-o-w** [comment]
I was just wondering the same thing. My library is on an SSD, so it’s surely not a read speed limitation.

The one thing I can imagine is that they’re throttling API requests to the various services that they use to get the metadata. It’s free after all and some other free tools (such as tiny media manager IIRC) make you register to get API keys yourself.

**Reply by Maltavius** [reply]
It's really slow even if you turn off all the metadata-fetchers and already have complete .nfo-files on your drive.

**Comment by ndom91** [comment]
This might be playing a part in it - it'll download ~200mb of each media file to scan it for various metadata. This can be adjusted with the `JELLYFIN_FFmpeg__analyzeduration` env var.

See: https://github.com/jellyfin/jellyfin/issues/9399

**Reply by s-h-a-d-o-w** [reply]
Maybe.
But I don't think your solution is acceptable for local use. Because unlike your scenario, jellyfin isn't really restricted there. And so it should just spawn as many ffprobes as possible (i.e. until space runs out) at the same time.
Maybe it doesn't do that. And/or maybe ffprobe can't max out a CPU core (which it really looks like because obviously, if it did, on my dual core machine, I would get consistent 50% usage even if it's not parallelized).
